testing/integration/proxy_url_test.go

```diff
@@ -341,6 +359,229 @@ func TestProxyURL(t *testing.T) {
+				require.NoError(t, err, "failed creating CA root")
+
+				caCertFile := filepath.Join(tmpDir, "ca.cert")
+				err = os.WriteFile(caCertFile, pair.Cert, 0o644&os.ModePerm)
+				require.NoError(t, err, "failed writing CA cert file %q", caCertFile)
+
+				caCertPool := x509.NewCertPool()
+				caCertPool.AddCert(caCert)
+
+				proxyCert, _, err := certutil.GenerateChildCert("localhost", []net.IP{net.IPv6loopback, net.IPv6zero, net.ParseIP("127.0.0.1")}, caKey, caCert)
+
+				// Create a fake proxy with TLS config to be used in fleet policy
+				proxyFleetPolicy := proxytest.New(t,
+					proxytest.WithRequestLog("proxy-fleet-policy", t.Logf),
+					proxytest.WithVerboseLog(),
+					proxytest.WithServerTLSConfig(&tls.Config{
+						RootCAs: caCertPool,
+						Certificates: []tls.Certificate{
+							{
+								Certificate: proxyCert.Certificate,
+								PrivateKey:  proxyCert.PrivateKey,
+								Leaf:        proxyCert.Leaf,
+							},
+						},
+					}))
+				err = proxyFleetPolicy.StartTLS()
+				require.NoError(t, err, "error starting TLS-enabled proxy")
+				t.Cleanup(proxyFleetPolicy.Close)
+
+				mockFleet.policyData.FleetProxyURL = &proxyFleetPolicy.LocalhostURL
+				mockFleet.policyData.SSL = &fleetservertest.SSL{
+					CertificateAuthorities: []string{caCertFile},
+					Renegotiation:          "never",
+				}
+				// now that we have fleet and the proxy running, we can add actions which
+				// depend on them.
+				action, err := fleetservertest.NewActionWithEmptyPolicyChange(
+					"actionID-TestValidProxyInThePolicy", *mockFleet.policyData)
+				require.NoError(t, err, "could not generate action with policy")
+
+				ackToken := "AckToken-TestValidProxyInThePolicy"
+				mockFleet.checkinWithAcker.AddCheckin(
+					ackToken,
+					0,
+					action,
+				)
+				return map[string]*proxytest.Proxy{"proxyFleetPolicy": proxyFleetPolicy}
+			},
+			enrollmentURL: func(fleet *fleetservertest.Server) string {
+				return fleet.LocalhostURL
+			},
+			wantErr: assert.NoError,
+			assertFunc: func(ctx context.Context, t *testing.T, fixture *integrationtest.Fixture, proxies map[string]*proxytest.Proxy, mockFleet *mockFleetComponents) {
+				// assert the agent is actually connected to fleet.
+				check.ConnectedToFleet(ctx, t, fixture, 5*time.Minute)
+
+				// ensure the agent is communicating through the proxy set in the policy
+				if !assert.Eventually(t, func() bool {
+					for _, r := range proxies["proxyFleetPolicy"].ProxiedRequests() {
+						if strings.Contains(
+							r,
+							fleetservertest.NewPathCheckin(mockFleet.policyData.AgentID)) {
+							return true
+						}
+					}
+
+					return false
+				}, 5*time.Minute, 5*time.Second) {
+					t.Errorf("did not find requests to the proxy defined in the policy")
+				}
+
+				inspectOutput, err := fixture.Exec(ctx, []string{"inspect"})
+				assert.NoError(t, err, "error running elastic-agent inspect")
+				t.Logf("elastic-agent inspect output:\n%s\n", string(inspectOutput))
```

Comment:
[Question]
Is it for debug?

Reply:
Yes, that would print out what the agent has stored as configuration... It's normally printed only on failed tests